### PR TITLE
range: fix warnings + unit test locale

### LINKED ISSUE #1407: range: compile warning, failed tests
compile warning:
```
/home/markus/Projekte/Elektra/current/src/plugins/range/range.c: In function ‘strToValue’:
/home/markus/Projekte/Elektra/current/src/plugins/range/range.c:97:1: note: the ABI of passing union with long double has changed in GCC 4.4
 {
 ^
/home/markus/Projekte/Elektra/current/src/plugins/range/range.c:115:51: warning: comparing floating point with == or != is unsafe [-Wfloat-equal]
   if (errno == ERANGE || (errno != 0 && v.Value.f == 0))
                                                   ^
```

failed test:
```
 88/126 Test  #94: testmod_range ....................***Failed    0.00 sec
RANGE     TESTS
==================

/home/markus/Projekte/Elektra/current/src/plugins/range/testmod_range.c:61: error in testFloat: failed
/home/markus/Projekte/Elektra/current/src/plugins/range/testmod_range.c:61: error in testFloat: failed

testmod_range RESULTS: 204 test(s) done. 2 error(s).
```

### PR DESCRIPTION
**Pull request by tom-wa**
# Purpose

fix #1407

# Checklist

- [x] commit messages are fine (with references to issues)
- [x] I ran all tests and everything went fine
- [ ] I added unit tests
- [ ] affected documentation is fixed
- [ ] I added code comments, logging, and assertions
- [ ] meta data is updated (e.g. README.md of plugins)

### CODE REVIEW

File: src/plugins/range/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,12 @@
 include (LibAddMacros)
 
+if (DEPENDENCY_PHASE)
+    set_source_files_properties(range.c PROPERTIES COMPILE_FLAGS -Wno-psabi)
+endif ()
+if(UNIX)
+    link_libraries(m)
```
Comment (markus2330) [inline src/plugins/range/CMakeLists.txt]: That is not a good idea, it adds the math library to *any* target added later.

Please use `LINK_LIBRARIES` in `add_plugin` instead.

File: tests/valgrind.suppression [hunk 2]
```diff
@@ -270,3 +270,10 @@
    fun:_ZN61kdbrestServicesConvertengineTest_FindSuitablePluginCheck_Test8TestBodyEv
    ...
 }
+{
+   <insert_a_suppression_name_here>
```
Comment (markus2330) [inline tests/valgrind.suppression]: Please say here that it is used because of testmod_range.c

File: src/plugins/range/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,12 @@
 include (LibAddMacros)
 
+if (DEPENDENCY_PHASE)
+    set_source_files_properties(range.c PROPERTIES COMPILE_FLAGS -Wno-psabi)
+endif ()
+if(UNIX)
```
Comment (markus2330) [inline src/plugins/range/CMakeLists.txt]: Properly done you would need to check with [CHECK_FUNCTION_EXISTS](http://stackoverflow.com/questions/32816646/can-cmake-detect-if-i-need-to-link-to-libm-when-using-pow-in-c) if libm is needed for fpclassify.

**Comment by markus2330** [comment]
See build server, it still seems to have memory issues: Memory checking results:
Uninitialized Memory Read - 14 https://build.libelektra.org/job/elektra-gcc-configure-debian-log/428/console

File: src/plugins/range/CMakeLists.txt (hunk 3 above)
Comment (tom-wa) [inline src/plugins/range/CMakeLists.txt]: check_function_exists keeps failing for fpclassify

**Comment by tom-wa** [comment]
jenkins build please

**Comment by markus2330** [comment]
Thank you!

**Pull request merged**